src/lib/notcurses.c

```diff
@@ -1650,7 +1650,7 @@ int ncplane_move_family_above(ncplane* restrict n, ncplane* restrict bpoint){
   // first, so that we know the topmost element of our new ensplicification.
   // at this point, n is the bottommost plane, and we're inserting above it.
   ncplane* targ = n;
-  while(above){
+  while(above && above != n){
```

Comment:
so above is initialized to `ncplane_above(n)` at the beginning of this function, which returns `n->above`. then we're moving `n` above `bpoint` using `ncplane_move_above()`, which splices `n` out from where it was, and places it above `bpoint` (which may be `NULL`, afaict). so at that point the following ought hold:

* `n->above` ought be whatever `bpoint->above` was, or `p->bottom` otherwise, or `NULL` if it's the only plane in the pile
* `n->below` ought be `bpoint`
* `bpoint->above` ought be `n`, unless `bpoint` was `NULL` in which case `p->bottom` now ought equal `n`
* if `p->top` was `n`, it is now `n->below`, unless `n` was the only plane in the pile, in which case it remains `n`
* `above` is whatever value `n->above` had when we started
* `below` is whatever value `n->below` had when we started

right? so if these postconditions have held, how can `above` ever equal `n` as we iterate from n to the top?

and even if we did, we still ought break the loop, since we have `tmp = ncplane_above(above);...above = tmp;`. so the only way this can hit an infinite loop is if there is truly a cycle in the loop as it results from `ncplane_move_above()`, in which case we have either a bug in `ncplane_move_above()`, or some unholy cyclic pile already existing.

so i think this is a bad fix, in that the real problem is elsewhere, wherever we're creating this cyclic pile. and we'd just run into problems elsewhere, since we'd need some guard against cycles everywhere we traverse the pile (which we do in several places, including shutdown).

so if you're truly hitting an infinite list here, i think the problem is probably in `ncplane_move_above()`. i'm not going to merge this PR, but if you're really seeing an infinite loop here, i think there's probably a real bug not far away.

Reply:
Here's a minimal program which will trigger this bug:

```C
int main(void)
{
	notcurses_options nc_opt = {0};
	struct notcurses *nc = notcurses_init(&nc_opt, NULL);

	ncplane_options plane_opt = { .rows = 1, .cols = 1 };
	struct ncplane *n = ncplane_create(notcurses_stdplane(nc), &plane_opt);
	struct ncplane *a = ncplane_create(n, &plane_opt);
	struct ncplane *b = ncplane_create(n, &plane_opt);
	struct ncplane *bpoint = ncplane_create(notcurses_stdplane(nc), &plane_opt);

	ncplane_move_family_above(n, bpoint);
	notcurses_stop(nc);
	return 0;
}
```
To illustrate what's happening here, before the call to `ncplane_move_above` we have:

    n a b bpoint
      ^above

Then `ncplane_move_above` moves `n` to the end. `above` still points to `a`:

    a b bpoint n
    ^above

After a few iterations of the loop, `above` points to `n`. `a` and `b` are moved to the top since they are descendants of `n`:

    b bpoint n a
    ^above

    bpoint n a b
    ^above

    bpoint n a b
           ^above

Then we proceed past `n`, move `a` to the back, `b` to the back, etc.

    bpoint n a b
             ^above

    bpoint n b a
             ^above

And here we have our infinite loop, with `a` and `b` swapping places forever.

Reply:
outstanding! i'm finishing up something for work but i'll look at this first thing tomorrow. thanks a lot for the details and especially the test case -- i'll turn that into a unit test.

very much appreciated!